### DISCUSSION TITLE
Major protocol version in cardano-nodel cluster framework.

### DISCUSSION THREAD

**Discussion by mikekeke** [cardano-foundation/cardano-wallet#3506] [Q&A] [answered]
Good day!
We are using extensively your awesome local cluster framework to build tooling for running and debugging contracts on reproducible disposable local network.

I've noticed, that current major protocol version is now hard-coded to `6` [here](https://github.com/input-output-hk/cardano-wallet/blob/404952e7943adbf589e7a80a9925e06dc1ad6c69/lib/shelley/src/Cardano/Wallet/Shelley/Launch/Cluster.hs#L917), even if the network uses Babbage era. Is there a reason to not to bump it to `7` in case of `Babbage`?

The reason I'm asking is that we are testing contracts with V2 scripts now, and when on-chain code uses `PlutusTx.Builtins.serialiseData` we are getting error
```
Forbidden builtin function: (builtin serialiseData)
```

Bumping major version to `7` solves the issue, but for that to work we had to fork `cardano-wallet` and change `PParams`.

Maybe there is a way to override this version somehow during cluster setup?

**Comment by Anviking** [comment, marked as answer]
Hard-coded value bumped in #3518.
